packages/react-instantsearch-hooks/src/lib/useInstantSearchApi.ts

```diff
@@ -93,6 +93,11 @@ export function useInstantSearchApi<TUiState extends UiState, TRouteState>(
     }
 
     if (prevProps.searchClient !== props.searchClient) {
+      warn(
+        false,
```

Comment:
there should be some way to pass `yesIKnowMySearchClientIsntStable` to InstantSearch to disable the warning IMO